AG query; telegram channel scripts; clean up

martijnmartijnmartijnmartijn opened the pull request:
I added:
- An action network query to request all AG data. 
- A generalised function to add people to a telegram channel.
- Some function to convert Dutch geo data. 

I also cleaned up the codebase somewhat:
- Removed queries for the deprecated local_group field.
- Removed synchronisation of AG data to a google sheets file.

moritzcopter commented on rebel_management_utilities/members.py:
```diff
@@ -58,19 +58,11 @@ def get_custom_field(member, field):
-            if municipality in local_group_config['municipalities']:
-                local_group = _local_group
-                break
-    return local_group
```
I dont think we want to delete this block - otherwise we wont be able to find the local group of people who signed up before we introduced `municipalities`

martijnmartijnmartijnmartijn replied:
We actually converted everyones location data to a municipality to allow this, included the local_groups. I think Collin actually just finished this last week and deleted the local_group field altogether. I just queried everyone and no one has the field anymore, so we should be okay deleting this.

moritzcopter replied:
👍

moritzcopter commented on rebel_management_utilities/action_network.py:
```diff
@@ -76,16 +76,6 @@ def get_forms():
-        for local_group in get_config()['local_groups']:
-            if local_group in sender:
-                return local_group
-    return 'Other'
```
Any reason to delete this? This breaks the function below (line 93) - also this logic is still valid I think

martijnmartijnmartijnmartijn replied:
you're right, not sure what I was thinking - I added it in again.

moritzcopter commented on telegram_channels/sync_telegram_channels.py:
```diff
@@ -0,0 +1,96 @@
+                logs["no_telegram"].append(p)
+
+            except Error as e: # Other errors.
+                logs["other_error"].append(p)
```
Wouldnt it be easier to just log this (e.g. with `logging.warning`) or so?

martijnmartijnmartijnmartijn replied:
Yeah that's better. I added this

moritzcopter commented on telegram_channels/sync_telegram_channels.py:
```diff
@@ -0,0 +1,96 @@
+    contact = InputPhoneContact(
+        client_id=0,
+        phone=user["phone_number"],
+        first_name=user["given_name"],
```
`user` is not defined here no? - I think you mean `person`

martijnmartijnmartijnmartijn replied:
Fixed

moritzcopter commented on rebel_management_utilities/convert_geodata.py:
```diff
@@ -0,0 +1,55 @@
+        return coordinates
+    except AttributeError:
+        # invalid postcode if no coordinates are found
+        print(f'Coordinates not found: {postcode}')
```
`postcode` is not defined here no?

martijnmartijnmartijnmartijn replied:
Fixed this!

moritzcopter commented on telegram_channels/sync_telegram_channels.py:
```diff
@@ -0,0 +1,96 @@
+        - channel (string) : id of the telegram channel/group.
+        - invitation (string) : invitation message send to new people.
+    """
+    p_in_channel = [p.phone for user in await client.get_participants(channel) if p.phone]
```
`client` is not defined here

martijnmartijnmartijnmartijn replied:
Fixed

moritzcopter commented on telegram_channels/sync_telegram_channels.py:
```diff
@@ -0,0 +1,96 @@
+            except IndexError as e: # Person doesn't have Telegram.
+                logs["no_telegram"].append(p)
+
+            except Error as e: # Other errors.
```
I think you want to use `Exception` here

martijnmartijnmartijnmartijn replied:
Fixed

moritzcopter reviewed (changes requested):
Thanks for the Pull request @martijnmartijnmartijnmartijn !
There are a few bugs in the code which we should fix before merging.

Do you use an IDE for coding? These can highlight and help you find bugs like this very easily.  I can recommend PyCharm which I am using at work almost every day.

martijnmartijnmartijnmartijn commented:
@moritzcopter Thanks for the thorough review. I've now installed an IDE, and immediately the naming errors you mention ed popped up. I've always used a simple text editor so this is pretty damn nice haha. I think I fixed everything you requested now. 

In the mean time I added two more things:
- A function in members.py which can be used to update a persons data in Action Network.
- A function in action_network.py which computes an overview of the number of people in each local group, as well as in each municipality. I'm using it to create a map for outreach & training (check it out: https://www.datawrapper.de/_/coqXh/)

moritzcopter commented:
Thanks for the changes Martijn, that map looks pretty cool 👌  - I merged your changes
